sys/include/net/nanocoap.h

```diff
@@ -273,6 +279,38 @@ typedef struct {
     void *context;                  /**< ptr to user defined context data   */
 } coap_resource_t;
 
+/**
+ * @brief   CoAP resource request handler context
+ */
+struct _coap_resource_ctx {
+    const coap_resource_t *resource;    /**< resource of the request        */
+    void *context;                      /**< request context                */
+};
+
+/**
+ * @brief   Get resource path associated with a CoAP request
+ *
+ * @param[in]   ctx The request context
+ *
+ * @return  Resource path of the request
+ */
+static inline const char *coap_request_get_path(const coap_resource_ctx_t *ctx)
```

Comment:
I am not a big fan of `get_member()` functions. But they allow us to hide the details of `coap_resource_ctx_t` from the user, using just a forward declaration. Which is good to not cause API changes. Just having `coap_request_get_resource()` sufficient instead of  `... get_path()`, .`..get_methods()` and so on.